socorro/external/es/crashstorage.py

```diff
@@ -270,15 +267,15 @@ def get_index_for_crash(self, crash_date):
 
         return index
 
-    def save_processed_crash(self, raw_crash, processed_crash):
-        """Save processed crash report to Elasticsearch"""
-        # Massage the crash such that the date_processed field is formatted
-        # in the fashion of our established mapping.
-        reconstitute_datetimes(processed_crash)
+    def prepare_processed_crash(self, raw_crash, processed_crash):
```

Comment:
There are a bunch of subclasses that redact and adjust the data. Previously, these all overrode `save_processed_crash`, but they need to modify the data before calling that because that also indexes the final document.

This changes things so we have a prepare step first that everything can override. The order that prep steps happen isn't a big deal. After prep is done, then `save_processed_crash` kicks in.